src/modules/rlm_escape/rlm_escape.c

```diff
@@ -55,68 +63,86 @@ static char const hextab[] = "0123456789abcdef";
  *
  * @ingroup xlat_functions
  */
-static ssize_t escape_xlat(UNUSED TALLOC_CTX *ctx, char **out, size_t outlen,
-			   void const *mod_inst, UNUSED void const *xlat_inst,
-			   UNUSED request_t *request, char const *fmt)
+static xlat_action_t escape_xlat(TALLOC_CTX *ctx, fr_dcursor_t *out, request_t *request,
+				 void const *xlat_inst, UNUSED void *xlat_thread_inst,
+				 fr_value_box_list_t *in)
 {
-	rlm_escape_t const	*inst = mod_inst;
-	char const		*p = fmt;
-	char			*out_p = *out;
-	size_t			freespace = outlen;
-	size_t			len = talloc_array_length(inst->allowed_chars) - 1;
+	rlm_escape_t const	*inst;
+	void			*instance;
+	fr_value_box_t		*arg = fr_dlist_head(in);
+	char const		*p = arg->vb_strvalue;
+	char			*out_p;
+	size_t			reqlen = 0;
+	size_t			len;
+	fr_value_box_t		*vb;
+	fr_sbuff_t		sbuff;
+
+	memcpy(&instance, xlat_inst, sizeof(instance));
+	inst = talloc_get_type_abort(instance, rlm_escape_t);
```

Comment:
Instance data can be const too?

Reply:
Also all the complexity around utf8 chars can be removed.  That was just to stop truncation occurring in the middle of a char.  Now we have dynamically sized buffers it's not an issue